go/roothash/api/commitment/pool.go

```diff
@@ -157,6 +155,48 @@ func (p *Pool) ResetCommitments(round uint64) {
 	p.NextTimeout = TimeoutNever
 }
 
+func (p *Pool) prioritizeCommitments(commit *ExecutorCommitment) error {
+	// Prioritization is not necessary if there are no commitments in the pool
+	// or if the scheduler matches.
+	schedulerID, ok := p.Scheduler()
+	if !ok || schedulerID.Equal(commit.Header.SchedulerID) {
+		return nil
+	}
+
+	// Prevent placing commitments for different proposals during discrepancy resolution.
+	if p.Discrepancy {
+		logger.Debug("executor commitment with different scheduler cannot be placed during discrepancy resolution",
+			"round", p.Round,
+			"pool_scheduler_id", schedulerID,
+			"commit_scheduler_id", commit.Header.SchedulerID,
+		)
+		return ErrBadExecutorCommitment
+	}
+
+	// Reject commitments for proposals with lower priority.
+	poolPriority := p.Committee.SchedulingPriority(p.Round, schedulerID)
+	commitPriority := p.Committee.SchedulingPriority(p.Round, commit.Header.SchedulerID)
+
+	if poolPriority < commitPriority {
+		logger.Debug("executor commitment's priority too low",
+			"round", p.Round,
+			"pool_priority", poolPriority,
+			"commit_priority", commitPriority,
+		)
+		return ErrBadExecutorCommitment
+	}
+
+	// Drop commitments with lower priority and start building from scratch.
```

Comment:
I guess this means that any committee member can make the backup protocol fail by just pretending it computed a commitment based on a proposal from the higher-priority scheduler (since it doesn't need to present any proof that the higher-priority scheduler actually proposed something in this round?).

Reply:
That's true. As discussed offline, a proof is needed that the scheduler actually proposed a batch for that round.